pkg/virt-launcher/notify-client/client.go

```diff
@@ -318,13 +328,19 @@ func (n *Notifier) StartDomainNotifier(
 
 	// Run the event process logic in a separate go-routine to not block libvirt
 	go func() {
-		var interfaceStatuses []api.InterfaceStatus
-		var guestOsInfo *api.GuestOSInfo
+		data := domainEventData{
```

Comment:
I suggest you build it in one shot and not in stages. Stages are prone to errors.
Redefine the data at the call/actual usuage.

Reply:
you dont have all the parameter in the first place, and some of the values needs to be saved from one stage to another, but i can try to rebuilt it in the two needed places from scratch, saving the shared values between the two occurences,
lemme suggest the diff,
indeed at the moment there is a bug there.